### ISSUE TITLE
`needless_return` suggests empty block instead of void return

### ISSUE THREAD

**Issue by dswij** [rust-lang/rust-clippy#8177]
### Summary

`needless_return` suggests `{}` instead of `()` for empty returns in match statements

### Reproducer

I tried this code:

```rust
fn foo(x: bool) {
    match x {
        true => (),
        false => return, 
    }
}
```

I expected to see this happen: clippy suggests replacing return with `()`

Instead, this happened: clippy suggests replacing `return` with `{}`

```
warning: unneeded `return` statement
 --> src/main.rs:8:18
  |
8 |         false => return,
  |                  ^^^^^^ help: replace `return` with an empty block: `{}`
  |
  = note: `#[warn(clippy::needless_return)]` on by default
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#needless_return
```

### Version

```text
rustc 1.59.0-nightly (34926f0a1 2021-12-22)
binary: rustc
commit-hash: 34926f0a1681458588a2d4240c0715ef9eff7d35
commit-date: 2021-12-22
host: x86_64-unknown-linux-gnu
release: 1.59.0-nightly
LLVM version: 13.0.0
```

### Additional Labels

@rustbot label +L-suggestion
